build.psm1

```diff
@@ -4,6 +4,84 @@ $script:TestModulePathSeparator = [System.IO.Path]::PathSeparator
 
 $dotnetCLIChannel = "preview"
 $dotnetCLIRequiredVersion = "2.0.0-preview2-006502"
+$tagsUpToDate = $false
+function Get-PSTags
+{
+    param(
+        [Switch]
+        $AddRemoteIfMissing
+    )
+
+    $PowerShellRemoteUrl = "https://github.com/powershell/powershell.git"
+    $upstreamRemoteDefaultName = 'upstream'
+    $remotes = git remote
+    $upstreamRemote = $null
+    foreach($remote in $remotes)
+    {
+        $url = git remote get-url $remote
+        if($url -eq $PowerShellRemoteUrl)
+        {
+            $upstreamRemote = $remote
+            break
+        }
+    }
+
+    if(!$upstreamRemote -and $AddRemoteIfMissing.IsPresent -and $remotes -notcontains $upstreamRemoteDefaultName)
+    {
+        $null = git remote add $upstreamRemoteDefaultName $PowerShellRemoteUrl
+        $upstreamRemote = $upstreamRemoteDefaultName
+    }
+    elseif(!$upstreamRemote)
+    {
+        Write-Error "Please add a remote to PowerShell\PowerShell.  Example:  git remote add $upstreamRemoteDefaultName $PowerShellRemoteUrl" -ErrorAction Stop
+    }
+
+    git fetch --tags $upstreamRemote
+    $script:tagsUpToDate=$true
+}
+
+function Get-PSLatestTag
+{
+    if(!$tagsUpToDate)
+    {
+        Write-Warning "Run Get-PSTags to update tags"
+    }
+
+    $lastestTagCommitId = git rev-list --tags --max-count=1
+    return (git --git-dir="$PSScriptRoot/.git" describe $lastestTagCommitId)
+}
+
+function Get-PSVersion
+{
+    param(
+        [switch]
+        $OmitCommitId
+    )
+    if($OmitCommitId.IsPresent)
+    {
+        return (Get-PSLatestTag) -replace '^v'
+    }
+    else
+    {
+        return (Get-PSCommitId) -replace '^v'
+    }
+}
+
+function Get-PSCommitId
+{
+    # Not using git describe because it doesn't always get the latest tag
+    $latestTag = Get-PSLatestTag
+    # Get the last commit id in the current branch
+    $commitId = git log --pretty="%H" -n1   
+    # check in the branch is dirty
+    $isDirty = (git describe --dirty) -match '-dirty$'  
+    if($isDirty)
+    {
+        $commitId+='-dirty'
+    }
+
+    return "$latestTag-x-g$commitId"
```

Comment:
Maybe we continue to use `git describe` to get the commit id. It should work as long as we merge the release branch.